app/src/main/java/com/rkm/tasky/feature/loginscreen/viewmodel/LoginViewModel.kt

```diff
@@ -16,20 +21,29 @@ class LoginViewModel @Inject constructor(
 
     val email = TextFieldState()
     val password = TextFieldState()
-//    private val _isValidEmail = MutableStateFlow(isValidEmail(email.text.toString()))
-    private val _isValidEmail = MutableStateFlow(false)
-    val isValidEmail = _isValidEmail.asStateFlow()
+    val isValidEmail by derivedStateOf {
+        isValidEmail(email.text.toString())
+    }
     private val _showPassword = MutableStateFlow(false)
     val showPassword = _showPassword.asStateFlow()
 
 
     private fun isValidEmail(email: String): Boolean {
-       TODO("Not implemented yet")
+       return email.isNotEmpty() && Patterns.EMAIL_ADDRESS.matcher(email).matches()
```

Comment:
Here, you're using an Android-specific dependency (as you can see in the imports `android.util.Patterns`) which doesn't really belong in the ViewModel. Instead, I'd recommend to create an abstraction for this to keep this class locally unit testable.

Every time, you have implementation details in domain or presentation and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface) in domain
2. You write an implementation of that in data (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing it in domain
4. Now you can decide which implementation you pass when you create that class/use case
For example:

*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your use case and use there.